tests/test_generic.py

```diff
@@ -31,6 +31,41 @@ def test_season(self, q_series):
         assert o[0] == 31 + 29
 
 
+class TestSelectRollingResampleOp:
+    def test_rollingmax(self, q_series):
+        q = q_series(np.arange(1, 366 + 365 + 365 + 1))  # 1st year is leap
+        o = generic.select_rolling_resample_op(
+            q, "max", window=14, window_center=False, window_op="mean"
+        )
+        np.testing.assert_array_equal(
+            [
+                np.mean(np.arange(353, 366 + 1)),
+                np.mean(np.arange(353 + 365, 366 + 365 + 1)),
+                np.mean(np.arange(353 + 365 * 2, 366 + 365 * 2 + 1)),
+            ],
+            o.values,
+        )
+
+    def test_rollingmaxindexer(self, q_series):
+        q = q_series(np.arange(1, 366 + 365 + 365 + 1))  # 1st year is leap
+        o = generic.select_rolling_resample_op(
+            q, "min", window=14, window_center=False, window_op="max", season="DJF"
+        )
+        np.testing.assert_array_equal(
+            [14, 367, 367 + 365], o.values
+        )  # 14th day for 1st year, then Jan 1st for the next two
+
+    def test_freq(self, q_series):
+        q = q_series(np.arange(1, 366 + 365 + 365 + 1))  # 1st year is leap
+        o = generic.select_rolling_resample_op(
+            q, "max", window=3, window_center=True, window_op="sum", freq="MS"
+        )
+        np.testing.assert_array_equal(
+            [np.sum([30, 31, 32]), np.sum([30 + 29, 31 + 29, 32 + 29])],
+            o.isel(time=slice(0, 2)).values,
+        )
```

Comment:
Could you add an assertion to ensure the output units are what you expected ?

Original q is in "m3 s-1", after the rolling sum it is now "m3" and then after the max it should still be "m3".

I think this doesn't work. `to_agg_units` doesn't cover the case where the original units have a "temporal" dimension before a sum.